langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockEmbeddingIT.java

```diff
@@ -116,6 +117,42 @@ void testBedrockCohereEmbedEnglishTextV3() {
         assertThat(embeddingModel.dimension()).isEqualTo(1024);
     }
 
+    @Test
+    void testBedrockCohereEmbedEnglishTextV3MultipleTextSegments() {
+
+        BedrockCohereEmbeddingModel embeddingModel = BedrockCohereEmbeddingModel
+            .builder()
+            .region(Region.US_EAST_1)
+            .maxRetries(1)
+            .model(BedrockCohereEmbeddingModel.Types.CohereEmbedEnglishTextV3.getValue())
+            .build();
+
+        assertThat(embeddingModel).isNotNull();
+
+        List<TextSegment> segments = new ArrayList<>();
+        segments.add(TextSegment.from("Hello world!"));
+        segments.add(TextSegment.from("How are you?"));
+
+        Response<List<Embedding>> response = embeddingModel.embedAll(segments);
+        assertThat(response).isNotNull();
+
+        List<Embedding> embeddings = response.content();
+        assertThat(embeddings).hasSize(1);
```

Comment:
```suggestion
        assertThat(embeddings).hasSize(2);
```

It should be 2. 2 segments = 2 embeddings

Reply:
Then there is the issue. Have a look at the base interface `BedrockEmbeddingResponse.java`. It has method `Embedding toEmbedding();`
Now if you look first implementation: `BedrockTitanEmbeddingResponse.java` it has only this method 
`    @Override
    public Embedding toEmbedding() {
        return new Embedding(embedding);
    }
`

And finally if you look at `AbstractBedrockEmbeddingModel.java` ln:55-58, 
`
        for (T response : responses) {
            embeddings.add(response.toEmbedding());
            totalInputToken += response.getInputTextTokenCount();
        }
`

But the "problem" with Cohere is that return Json is `float[][] embeddings` and not `float[] embedding` 
So there are no 2 responses, just 1. but there are 2 response.embeddings (response.embeddings[0] and response.embeddings[1]) and if I need to change Abstract class, then previous `BedrockTitanEmbeddingResponse.java` will stop working :( 

Any advice is more than welcomed.

Reply:
@staleks we can change `BedrockEmbeddingResponse` to support multiple `Embedding`s. I can't commit to your branch, could you please apply this patch?
```
Subject: [PATCH] support multiple embeddings in BedrockEmbeddingResponse
---
Index: langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingResponse.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingResponse.java b/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingResponse.java
--- a/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingResponse.java	(revision 9db580f0d90794294540a92daf791148b5248592)
+++ b/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingResponse.java	(revision 161d7925ac8589f844a80defd5bd4d3bc5b7ef46)
@@ -3,6 +3,7 @@
 import dev.langchain4j.data.embedding.Embedding;
 import dev.langchain4j.model.bedrock.internal.BedrockEmbeddingResponse;
 
+import java.util.Arrays;
 import java.util.List;
 
 /**
@@ -11,19 +12,18 @@
 class BedrockCohereEmbeddingResponse implements BedrockEmbeddingResponse {
 
     private String id;
-
     private List<String> texts;
-
     private float[][] embeddings;
-
     private int inputTextTokenCount = 0;
 
     @Override
-    public Embedding toEmbedding() {
+    public List<Embedding> toEmbeddings() {
         if (embeddings == null) {
-            return null;
+            return List.of();
         }
-        return new Embedding(embeddings[0]);
+        return Arrays.stream(embeddings)
+            .map(Embedding::from)
+            .toList();
     }
 
     public String getId() {
Index: langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockTitanEmbeddingResponse.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockTitanEmbeddingResponse.java b/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockTitanEmbeddingResponse.java
--- a/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockTitanEmbeddingResponse.java	(revision 9db580f0d90794294540a92daf791148b5248592)
+++ b/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockTitanEmbeddingResponse.java	(revision 161d7925ac8589f844a80defd5bd4d3bc5b7ef46)
@@ -5,6 +5,8 @@
 import lombok.Getter;
 import lombok.Setter;
 
+import java.util.List;
+
 /**
  * Bedrock Titan embedding response
  */
@@ -16,7 +18,7 @@
     private int inputTextTokenCount;
 
     @Override
-    public Embedding toEmbedding() {
-        return new Embedding(embedding);
+    public List<Embedding> toEmbeddings() {
+        return List.of(Embedding.from(embedding));
     }
 }
Index: langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractBedrockEmbeddingModel.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractBedrockEmbeddingModel.java b/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractBedrockEmbeddingModel.java
--- a/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractBedrockEmbeddingModel.java	(revision 9db580f0d90794294540a92daf791148b5248592)
+++ b/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractBedrockEmbeddingModel.java	(revision 161d7925ac8589f844a80defd5bd4d3bc5b7ef46)
@@ -54,7 +54,7 @@
         int totalInputToken = 0;
         final List<Embedding> embeddings = new ArrayList<>();
         for (T response : responses) {
-            embeddings.add(response.toEmbedding());
+            embeddings.addAll(response.toEmbeddings());
             totalInputToken += response.getInputTextTokenCount();
         }
 
Index: langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/BedrockEmbeddingResponse.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/BedrockEmbeddingResponse.java b/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/BedrockEmbeddingResponse.java
--- a/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/BedrockEmbeddingResponse.java	(revision 9db580f0d90794294540a92daf791148b5248592)
+++ b/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/BedrockEmbeddingResponse.java	(revision 161d7925ac8589f844a80defd5bd4d3bc5b7ef46)
@@ -2,17 +2,14 @@
 
 import dev.langchain4j.data.embedding.Embedding;
 
+import java.util.List;
+
 /**
  * Bedrock embedding response
  */
 public interface BedrockEmbeddingResponse {
 
-    /**
-     * Get embedding
-     *
-     * @return embedding
-     */
-    Embedding toEmbedding();
+    List<Embedding> toEmbeddings();
 
     /**
      * Get input text token count
Index: langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingResponseTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingResponseTest.java b/langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingResponseTest.java
--- a/langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingResponseTest.java	(revision 9db580f0d90794294540a92daf791148b5248592)
+++ b/langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingResponseTest.java	(revision 161d7925ac8589f844a80defd5bd4d3bc5b7ef46)
@@ -5,9 +5,9 @@
 import java.util.List;
 import java.util.UUID;
 
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
-import static org.junit.jupiter.api.Assertions.assertNull;
 
 class BedrockCohereEmbeddingResponseTest {
 
@@ -15,15 +15,16 @@
     void testToEmbeddingNull() {
         BedrockCohereEmbeddingResponse response = new BedrockCohereEmbeddingResponse();
         response.setEmbeddings(null);
-        assertNull(response.toEmbedding());
+        assertThat(response.toEmbeddings()).isEmpty();
     }
 
     @Test
     void testToEmbeddingNotNull() {
         BedrockCohereEmbeddingResponse response = new BedrockCohereEmbeddingResponse();
-        response.setEmbeddings(new float[][]{{1.0f, 2.0f, 3.0f}});
-        assertNotNull(response.toEmbedding());
-        assertEquals(1.0f, response.toEmbedding().vector()[0]);
+        response.setEmbeddings(new float[][]{{1.0f, 1.0f, 1.0f}, {2.0f, 2.0f, 2.0f}});
+        assertThat(response.toEmbeddings()).hasSize(2);
+        assertThat(response.toEmbeddings().get(0).vector()[0]).isEqualTo(1.0f);
+        assertThat(response.toEmbeddings().get(1).vector()[0]).isEqualTo(2.0f);
     }
 
     @Test
Index: langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockEmbeddingIT.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockEmbeddingIT.java b/langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockEmbeddingIT.java
--- a/langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockEmbeddingIT.java	(revision 9db580f0d90794294540a92daf791148b5248592)
+++ b/langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockEmbeddingIT.java	(revision 161d7925ac8589f844a80defd5bd4d3bc5b7ef46)
@@ -84,6 +84,39 @@
         assertThat(embeddingModel.dimension()).isEqualTo(256);
     }
 
+    @Test
+    void testBedrockTitanEmbeddingModelV2MultipleSegments() {
+
+        // given
+        BedrockTitanEmbeddingModel embeddingModel = BedrockTitanEmbeddingModel.builder()
+            .model(BedrockTitanEmbeddingModel.Types.TitanEmbedTextV2.getValue())
+            .build();
+
+        List<TextSegment> segments = List.of(
+            TextSegment.from("Hello world!"),
+            TextSegment.from("How are you?")
+        );
+
+        // when
+        Response<List<Embedding>> response = embeddingModel.embedAll(segments);
+
+        // then
+        List<Embedding> embeddings = response.content();
+        assertThat(embeddings).hasSize(2);
+
+        Embedding embedding = embeddings.get(0);
+        assertThat(embedding.vector()).hasSize(1024);
+
+        TokenUsage tokenUsage = response.tokenUsage();
+        assertThat(tokenUsage.inputTokenCount()).isEqualTo(9);
+        assertThat(tokenUsage.outputTokenCount()).isNull();
+        assertThat(tokenUsage.totalTokenCount()).isEqualTo(9);
+
+        assertThat(response.finishReason()).isNull();
+
+        assertThat(embeddingModel.dimension()).isEqualTo(1024);
+    }
+
     @Test
     void testBedrockCohereEmbedEnglishTextV3() {
 
@@ -137,7 +170,7 @@
         assertThat(response).isNotNull();
 
         List<Embedding> embeddings = response.content();
-        assertThat(embeddings).hasSize(1);
+        assertThat(embeddings).hasSize(2);
 
         Embedding embedding = embeddings.get(0);
         assertThat(embedding.vector()).hasSize(1024);

```

Reply:
done